### DISCUSSION TITLE
Some Folders show thumbnail but no contents

### DISCUSSION THREAD

**Discussion by elapse2039** [photoprism/photoprism#1897] [Q&A] [answered]
**What does not work as expected?**

Some but not all Folders appear empty after a complete re-scan.  
The photos from these Folders can be seen in Search or Calendar views.

**How can we reproduce it?**

Rebuild library.

**What behavior do you expect?**

The Project folder should be populated with images.

**Can you provide us with example files for testing or screenshots?**

A sample JPG file has been included.  They were all previously successfully viewable in an earlier version of Photoprism.

**What version you are using?**

I upgraded to 211215 and manually removed all the JSON files in the originals sub folders as these were turning up with "JSON" icons.

**Any other helpful information?**

I happened to discover this when I was converting some Folders into Albums.  I noticed 15/16 were populated, then this "Dining Table Restoration" Folder had no images in it.  All other Folders (so far) have images and successfully generate Albums from them.

"Project - Dining Table Restoration" is found and present in Folder view...
![screenshot977](https://user-images.githubusercontent.com/5325698/148257161-956f860d-9776-47c6-b5ed-9351b1d0bc67.png)

Entering that folder...
![screenshot978](https://user-images.githubusercontent.com/5325698/148256518-ae797e05-45b7-4d9a-a607-bdba5b4e71e5.png)

But you can find the photos from this folder via Search or Calendar...
![screenshot976](https://user-images.githubusercontent.com/5325698/148256564-3b1f333b-a8e3-4c2f-a7d1-47ad067b0a69.png)

I see no errors in the logs during a complete re-scan.  

Sample image file in the "Project - Dining Table Restoration" folder...
![IMG_9090](https://user-images.githubusercontent.com/5325698/148256775-913ca67c-e866-4549-a201-a3263559fb81.jpg)

**Comment by lastzero** [comment]
Are these stacks scattered across different folders? There's only one main folder possible, so files will be displayed there.

**Comment by elapse2039** [comment]
There is one folder which contains the original photos and I can see these files via SFTP/local command line.  
It is empty when I view it via Folders/.   (Not sure what a stack is?)

The Folder and the resulting Album I create both have nothing in them when viewed via PhotoPrism UX.  
When I do the same exact steps with other folders I can successfully create Albums and they are not empty.

**Comment by graciousgrey** [comment]
PhotoPrism stacks related files (see https://docs.photoprism.app/user-guide/organize/stacks/). But at least for the image with the white table that's not the case as I see on the screenshot that it consists of onlye one file.

Is it possible that you have duplicates of those images in another folder?

Could you check the name that is shown on the files tab (https://docs.photoprism.app/user-guide/organize/edit/#edit-files) for the images that should be in the empty folder?

**Comment by elapse2039** [comment]
If I search for one of the dining table photos, I can then look at the FILES tab and see...

![screenshot980](https://user-images.githubusercontent.com/5325698/148396199-8a02efd4-5432-48e7-ac5b-09acf1bf344a.png)

When you say duplicates, you mean filename duplicate?

I see there are two files with the same name... is this a problem?  Or is uniqueness defined by path and file name?

/mnt/PHOTOPRISM/photoprism/originals$ find . -name "IMG_9190.jpg"
./gphotos/2018/20180401/IMG_9190.jpg
./gphotos/projects/Project - Dining Table restoration /IMG_9190.jpg

**Comment by lastzero** [comment]
No, SHA-1 duplicates (same content with extremely high probability):

https://en.wikipedia.org/wiki/SHA-1

**Reply by elapse2039** [reply]
Arh, good to know. Ok, there are two files with same name (IMG_9190.jpg) but they are completely different photos.
So as far as I can tell there is only one file that has the sha1 of IMT_9190.jpg in Dining Table Restoration folder.

4e6f523cf4d7798a73004e26338539ae0912f126  ./gphotos/projects/Project - Dining Table restoration /IMG_9160.jpg
(This matches the Photoprism photo FILES details "Hash".)

I sha'd all the files in photoprism originals/ folders (~30k images & 1k videos) and there is only one entry for this IMG_9160.jpg file.

**Comment by lastzero** [comment]
I'll move this to discussions to determine the cause - like duplicates that get ignored - before we add it to our issue tracker as a confirmed bug :)

**Comment by graciousgrey** [comment, marked as answer]
Could you enable debug logs and then start a complete rescan of only one of the empty folders and check what the logs say?

**Reply by elapse2039** [reply]
![screenshot985](https://user-images.githubusercontent.com/5325698/148705517-0a90988f-5687-4fa6-89b8-fcd98daf82c5.png)
[sample.txt](https://github.com/photoprism/photoprism/files/7836194/sample.txt)

sample debug logs cut and paste from GUI.  Let me know if there's a log file somewhere I should grab?

**Reply by graciousgrey** [reply]
It looks like your folder name is ending with a space` "Project - Dining Table restoration ".` This might be the issue. 
Could you remove the space (`"Project - Dining Table restoration"`) from the folder name and do a rescan again?

**Reply by davepfrommer** [reply]
I'm having the same issue with several of my folders as well.

I can confirm that there is no trailing space on the folders that show a thumbnail, but show an empty folder when clicked.

**Reply by lastzero** [reply]
Then it may have other reasons. Also note folder thumbs are cached, so they can show something if contents only changed recently.

**Reply by davepfrommer** [reply]
@lastzero do you mean cached at the browser level or within the application? If the former, I did test with different browers and caching disabled.

**Reply by lastzero** [reply]
Both is possible, although the server cache should get invalidated in critical cases like when you change private flags.

**Reply by elapse2039** [reply]
It does indeed have a space...
...
'Project - Bunk Ladders'
'Project - Chess Board'
'Project - Dining Table restoration '
'Project - Front porch'
'Project - Hot Tub'
'Project - Kitchen Garden'
'Project - Maple Goblets'
...

Removing the space and rescan fixed the issue!  And I can create a non-empty Album.  I must have added a space accidentally using Google Photos app and exported as a directory name including that trailing space. 

Thanks for your keen eyes - I did not spot that!

**Reply by davepfrommer** [reply]
@lastzero I've re-indexed several directories in the originals folder which have the same behavior; i.e, thumbnail on a folder, but 0 pictures displayed when clicked.

I've looked at the permissions, folder names, spaces, file names of images within the folder.  I have also turned on debug logs.

Unfortunately, I cannot determine any differences that may cause an issue.

Here is an example of the log file when clicking on one of the folders:


2022-01-10 20:52:24 DEBU http: GET /api/v1/photos?count=60&offset=0&album=ar5i06z2rt4q956a&filter=path:%2205-12-2013+Father%27s+Day+at+Grandpas%22+public:true&merged=true&country=&camera=0&order=added&q= (200) [1.955309ms]
2022-01-10 20:52:24 DEBU photos: found 0 results for filter:"path:05-12-2013 Father's Day at Grandpas public:true" path:"05-12-2013 Father's Day&Grandpas" public:true dist:20 album:ar5i06z2rt4q956a count:60 order:added merged:true [1.718661ms]
2022-01-10 20:52:24 DEBU http: GET /api/v1/albums/ar5i06z2rt4q956a (200) [4.518184ms]
2022-01-10 20:52:20 DEBU http: GET /api/v1/t/45405cb3daf4c3401827e9be13a8bda1a2934728/9b72b8df/tile_500 (200) [1.807319ms]
2022-01-10 20:52:20 ERRO thumb: thumbnail not cached
2022-01-10 20:52:20 DEBU http: GET /api/v1/t/660c8fdc0dd74f933147387bb61961abe9f5b073/9b72b8df/tile_500 (200) [2.037978ms]
2022-01-10 20:52:20 ERRO thumb: thumbnail not cached

Thoughts?

**Reply by davepfrommer** [reply]
@lastzero actually, I believe I've found something..

It seems like all the folders that have the word "in" or "at" are been converted/escaped.

For example, clicking on the above folder, I see this:
2022-01-10 21:07:53 DEBU photos: found 0 results for filter:"path:05-12-2013 Father's Day at Grandpas public:true" path:"05-12-2013 Father's Day&Grandpas" public:true dist:20 album:ar5i06z2rt4q956a count:60 order:added merged:true [1.559439ms]

05-12-2013 Father's Day&Grandpas instead of 05-12-2013 Father's Day at Grandpas 

Thoughts?

**Reply by graciousgrey** [reply]
@davepfrommer good catch! The content of folders with names including spaces in combination with words like "in" or "at" cannot be displayed in the folder section. I will create an issue to fix this

**Reply by graciousgrey** [reply]
@davepfrommer this should be fixed with the next update (later today or tomorrow) :)
